File: docs/coding/samples/numcal.cs
using System;
using System.Collections.Generic;

class NumberCalculator
{
    static void Main()
    {
        Console.WriteLine("=== Number Calculator ===");
        
        List<double> numbers = new List<double>();
        
        Console.WriteLine("Enter numbers (type 'done' to finish):");
        
        while (true)
        {
            Console.Write("> ");
            string input = Console.ReadLine()?.Trim();
            
            if (string.IsNullOrEmpty(input)) continue;
            
            if (input.Equals("done", StringComparison.OrdinalIgnoreCase))
            {
                break;
            }
            
            if (double.TryParse(input, out double num))
            {
                numbers.Add(num);
            }
            else
            {
                Console.WriteLine($"'{input}' is not a valid number. Try again.");
            }
        }
        
        if (numbers.Count == 0)
        {
            Console.WriteLine("No numbers entered.");
            return;
        }
        
        double sum = 0;
        double max = double.MinValue;
        double min = double.MaxValue;
        
        foreach (double num in numbers)
        {
            sum += num;
            if (num > max) max = num;
            if (num < min) min = num;
        }
        
        double average = sum / numbers.Count;
        
        Console.WriteLine("\nResults:");
        Console.WriteLine($"Count: {numbers.Count}");
        Console.WriteLine($"Sum: {sum:F2}");
        Console.WriteLine($"Average: {average:F2}");
        Console.WriteLine($"Max: {max:F2}");
        Console.WriteLine($"Min: {min:F2}");
        
        Console.WriteLine($"\nNumbers entered: [{string.Join(", ", numbers)}]");
    }
}

File: docs/coding/samples/numcal.rs
use std::io;

fn main() {
    println!("=== Number Calculator ===");
    
    let mut numbers: Vec<f64> = Vec::new();
    
    println!("Enter numbers (type 'done' to finish):");
    
    loop {
        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read line");
        
        let input = input.trim();
        if input.eq_ignore_ascii_case("done") {
            break;
        }
        
        match input.parse::<f64>() {
            Ok(num) => numbers.push(num),
            Err(_) => println!("'{}' is not a valid number. Try again.", input)
        }
    }
    
    if numbers.is_empty() {
        println!("No numbers entered.");
        return;
    }
    
    let sum: f64 = numbers.iter().sum();
    let average = sum / numbers.len() as f64;
    let max = numbers.iter().fold(f64::NEG_INFINITY, |a, &b| a.max(b));
    let min = numbers.iter().fold(f64::INFINITY, |a, &b| a.min(b));
    
    println!("\nResults:");
    println!("Count: {}", numbers.len());
    println!("Sum: {:.2}", sum);
    println!("Average: {:.2}", average);
    println!("Max: {:.2}", max);
    println!("Min: {:.2}", min);
    
    println!("\nNumbers entered: {:?}", numbers);
}